substrate/client/authority-discovery/src/worker.rs

```diff
@@ -485,6 +505,11 @@ where
 				debug!(target: LOG_TARGET, "Successfully put hash '{:?}' on Dht.", hash)
 			},
 			DhtEvent::ValuePutFailed(hash) => {
+				if !self.latest_published_kad_keys.remove(&hash) {
+					// Not a value we have published or received multiple times.
+					return;
+				}
+
```

Comment:
Don't you want to reset the timer here? I mean given what you write in the pr description, I got the impression.

Reply:
Yep, that makes sense!

I reset the `publish_interval` here. I'm a bit concerned about being too aggressive when we constantly fail to put the DHT records. We'd reset and resubmit the DHT records at 2 seconds intervals until all records succeed.

This would mean that for freshly-started node (where failure might be high) we'd resubmit at 2 seconds internvals instead of exponentially increasing ones 🤔

Reply:
Just speculating: may be introduce another "ephemeral" (i.e., `Option`) exponential-backoff timer just to handle failed publishings?